website/docs/core-concepts/connecting-to-data-sources/connecting-to-databases.md

```diff
@@ -5,59 +5,78 @@ description: >-
+* Rename and save your datasource.
+
+
+
+Once your datasource has been added successfully, a success pop-up appears at the top. 
+
+### Mock database
+
 
-**18.223.74.85** and **3.131.104.27** are the IP addresses of the Appsmith cloud instances that need to be whitelisted.
+Appsmith provides two sample datasets to help you connect the data and test the platform. 
+
+1. A sample MongoDB database with a collection called "movies."
+2. A sample PostgreSQL database with a table called "users."
+
+:::note
+The mock databases are public and shared by all users, which may result in mangled data. To refresh the data, the databases are automatically **reset every day**, so any updates made to these databases are temporary.
 :::
 
-Once the appsmith instance has been [whitelisted](../../learning-and-resources/how-to-guides/aws-whitelist.md):
+<VideoEmbed host="youtube" videoId="TrV8h_Dvhbg" title="Using A Sample Database " caption="How to use mock database"/>
 
-1. Click on **+** next to **Datasources**.
-2. You’ll see a list of Datasources that Appsmith can connect to.
-3. Choose a Database from the [supported list](connecting-to-databases.md#supported-databases)
-4. Provide the configuration details required to connect to your database. You may need to contact your database admin to [whitelist appsmith cloud](../../learning-and-resources/how-to-guides/aws-whitelist.md)
-5. Click on **Test** to verify that Appsmith is able to connect to your database using the details provided by you.
-6. **Save** your **Datasources**.
+* Click on **+** next to Datasources.
+* Select the mock **users** PostgreSQL database.
+* You can create queries by selecting the **New API +** button available on the **datasource** page.
+* Write the query in the query editor and click the **Run** button to execute it. For example:
 
-![](</img/create_ds.gif>)
+```sql
+SELECT * FROM users;
+```
+The select query retrieves the data from the users' table. You can display the query results in a Table widget by [binding the data](/core-concepts/data-access-and-binding/displaying-data-read#displaying-data-in-a-widget) in the Table Data property.
 
-:::tip
+:::note
 Databases configured within one page of an app can be accessed by developers anywhere within the workspace.
 :::
 
-## General Notes
+## General notes
+Appsmith establishes a connection pool with the database server when you first connect the database to your app. This allows subsequent queries to be executed efficiently. If an idle connection is closed by the database server, Appsmith creates a new connection for the next query.
+
+For certain database plugins such as PostgreSQL and MySQL, Appsmith maintains a connection pool because multiple queries can't be run on a single connection.
+
+The maximum number of concurrent queries that can be run on a database with Appsmith is 5. If the application tries to execute more concurrent queries, you may receive an error message stating that the connection isn't available.
```

Comment:
This is not correct, please remove this. We can instead mention that we hold max 5 connections in the pool. The limit of `5` is a limit on the number of max connections in a pool. `5` connections in a pool does not translate to `5` max concurrent queries. The number of concurrent queries can be much larger.

Reply:
updated